core/commands/stat.go

```diff
@@ -147,35 +151,31 @@ Example:
 		}
 	},
 	Type: metrics.Stats{},
-	PostRun: cmds.PostRunMap{
-		cmds.CLI: func(res cmds.Response, re cmds.ResponseEmitter) error {
-			polling, _ := res.Request().Options[statPollOptionName].(bool)
+	Encoders: cmds.EncoderMap{
+		cmds.Text: cmds.MakeEncoder(func(req *cmds.Request, w io.Writer, v interface{}) error {
+			polling, _ := req.Options[statPollOptionName].(bool)
 
 			if polling {
-				fmt.Fprintln(os.Stdout, "Total Up    Total Down  Rate Up     Rate Down")
+				// print only once in `--poll` flag
+				once.Do(func() {
```

Comment:
This won't work for multiple invocations of the command, not sure what's the best way to do that. cc @keks

Reply:
The proper way to do this is using `PostRun`. Encoders are for processing individual elements, while `PostRun` is passed the stream that is emitted by `Run` and can do transport (e.g. cli or http) specific presentation.

Reply:
To clarify: I think we need a second `PostRunFunc` in the `PostRun` map with index `cmds.JSON`.